dev_guide/builds.adoc

```diff
@@ -598,6 +598,16 @@ containing a Dockerfile that overwrites any Dockerfile that may exist in the
 source repository.
 ====
 
+When using the Git repository as a source without specifying the `*ref*`
+field, OpenShift will perform a shallow clone (`--depth=1` clone). That means
+only the `HEAD` (usually the `master` branch) will be downloaded. This result
+into much faster download times of repositories with rich history of commits.
+
+The shallow clone is also used when the `*ref*` field is specified and set to an
+existing remote branch name. However, if you specify the `*ref*` field to a commit
+SHA256 we have to fallback into a regular git clone operation and then checkout the
+commit, because using the `--depth=1` option only works with named branch refs.
```

Comment:
s/we have to fallback into/the system will fallback to a regular..../

also add a sentence about how you can specify "master" as your ref to force a deep clone if your build logic needs access to the commit history.